modules/programs/ssh.nix

```
@@ -60,10 +60,20 @@ let
   matchBlockModule = types.submodule ({ dagName, ... }: {
     options = {
       host = mkOption {
-        type = types.str;
+        type = types.nullOr types.str;
+        default = null;
         example = "*.example.org";
         description = ''
-          The host pattern used by this conditional block.
+          `Host` pattern used by this conditional block. See `ssh_config(5)` for `Host` block details. `match` option takes precedence if defined.
+        '';
+      };
+
+      match = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "host <hostname> canonical\nhost <hostname> exec \"ping -c1 -q 192.168.17.1\"";
+        description = ''
+          `Match` block conditions used by this block. See `ssh_config(5)` for `Match` block details. `match` option takes precedence over `host` if defined.
```

```suggestion
          <literal>Match</literal> block conditions used by this block. See
           <citerefentry>
             <refentrytitle>ssh_config</refentrytitle>
             <manvolnum>5</manvolnum>
           </citerefentry>
           for <literal>Match</literal> block details.
           This option takes precedence over
           <option>ssh.matchBlocks.*.match<option>
           if defined.
```

Not quite sure what the actual option name is, please correct if necessary.